### PR TITLE
Bug Tracking

### PR DESCRIPTION
**Pull request by SaranjeetKaur**
This chapter describes how bug tracking works in R and its related documentations.

### CODE REVIEW

File: 11-bug_tracking.Rmd
Comment (llrs) [inline 11-bug_tracking.Rmd:81]: I had the impression that the repo r-devel/r-svn was not officially supported as a place to report bugs. It is only a community effort to make it easier to submit patches. Has this changed? Is there any reference to support this?

File: 11-bug_tracking.Rmd
Comment (llrs) [inline 11-bug_tracking.Rmd:89]: If there isn't a second paragraph, I don't think it makes sense to enumerate them.  Also I would merge the #RCorePkgBug and the #RLangBug section in one. To distinguish between R-core and community supported packages.

File: 11-bug_tracking.Rmd
Comment (llrs) [inline 11-bug_tracking.Rmd:95]: Just tested this. Using `bug.report("MASS")` on RStudio doesn't work as expected on RStudio. However if one uses `utils::bug.report(package = "MASS")` it works.

File: 11-bug_tracking.Rmd
Comment (llrs) [inline 11-bug_tracking.Rmd:97]: If this paragraph is for #nonRCorePkgBug, #RCorePkgBug and the #RLangBug section it should be on a different section or out of this. Perhaps it could be expanded to explain how to ensure that the bug is not already reported or fixed and what to do in those occasions that it is already solved/reported. Some suggestions, search error or function name not the whole code.

File: 11-bug_tracking.Rmd
Comment (llrs) [inline 11-bug_tracking.Rmd:104]: Not sure how agnostic this book should be but perhaps mention tools like reprex, or other documentation about how to make a minimal reproducible example ? The great [SO question](https://stackoverflow.com/q/5963269/2886003)

File: 11-bug_tracking.Rmd
Comment (llrs) [inline 11-bug_tracking.Rmd:105]: This section seems only for #RCorePkgBug and the #RLangBug, perhaps it would be better to add a note at the beginning of the section? (Not that for some other packages this information is not relevant but as the links just point to r-project better to be explicit)

File: 11-bug_tracking.Rmd
Comment (llrs) [inline 11-bug_tracking.Rmd:110]: When I first submitted a bug I didn't know the Severity to give to a bug or the Component to use, perhaps some paragraphs about each element of a bug report will make it easier for people to understand how to make a good bug report on Bugzilla

File: 11-bug_tracking.Rmd
Comment (llrs) [inline 11-bug_tracking.Rmd:117]: Perhaps mention that even if an issue or bug report is closed one can still comment on it.

**Review by llrs — COMMENTED** [review]
I have made some comments of this PR. Hope this and the other PR can be merged soon on master.

File: 11-bug_tracking.Rmd
Comment (SaranjeetKaur) [reply 11-bug_tracking.Rmd:55]: Fixing bugs will be discussed in the chapter on lifecycle of a patch. How to test bugs will probably be a part of the next chapter on reviewing bugs. Let's see how those chapters shape up!

File: 11-bug_tracking.Rmd
Comment (SaranjeetKaur) [reply 11-bug_tracking.Rmd:77]: In the introduction or as the last of the four sub-sub-sections of "Bug in ..."? What would be less confusing to readers? What do you suggest?

File: 11-bug_tracking.Rmd
Comment (SaranjeetKaur) [reply 11-bug_tracking.Rmd:83]: Yes, it is still a community effort to make submitting patches and/or issues easier.

File: 11-bug_tracking.Rmd
Comment (llrs) [reply 11-bug_tracking.Rmd:77]: I think that improvements on the book itself should be addressed on the introduction of the book (so on index.Rmd itself). It will be a great way to encourage more participation of people on the book.

File: 11-bug_tracking.Rmd
Comment (llrs) [reply 11-bug_tracking.Rmd:83]: Then I would change the wording to something like:


> 2. The bug report for $\textsf{R}$ Core-supported packages and/or their documentations must be submitted to $\textsf{R}$'s [Bugzilla](https://bugs.r-project.org/bugzilla/).

File: 11-bug_tracking.Rmd
Comment (llrs) [reply 11-bug_tracking.Rmd:55]: Ok, let's see. I hope that the different chapters can be merged soon, it will help to have a more general read of the book and see if the sections match well. Is there any plan on when to merge #3 or this PR?

Comment (SaranjeetKaur) [reply 11-bug_tracking.Rmd:55]: I hope it is merged soon.

**Comment by SaranjeetKaur** [comment]
> I have made some comments of this PR. Hope this and the other PR can be merged soon on master.

**Comment by SaranjeetKaur** [comment]
@llrs I've read all your comments. I am waiting for more reviews on this PR. Then I will start incorporating everyone's suggestions. Thank you so much for your patience!

File: 11-bug_tracking.Rmd
Comment (SaranjeetKaur) [reply 11-bug_tracking.Rmd:77]: Ok. My plan is to write the introduction to the book chapter after some chapters are completed. Yes, it sounds good to include the improvements to the book and reporting bugs in this documentation, in that chapter.

File: 11-bug_tracking.Rmd
Comment (willingc) [inline 11-bug_tracking.Rmd:24]:
```suggestion
## What condition might not be a bug?

```

File: 11-bug_tracking.Rmd
Comment (willingc) [inline 11-bug_tracking.Rmd:32]:
```suggestion
4. If some function is working, but it is not defined in the best generalised way, then consult someone to look over your code. This may perhaps not be a bug; instead, it might be an alternative way of writing the function.

```

File: 11-bug_tracking.Rmd
Comment (willingc) [inline 11-bug_tracking.Rmd:71]:
```suggestion
Once you confirm a bug exists, you need to submit a bug report so that it gets fixed.

```

File: 11-bug_tracking.Rmd
Comment (willingc) [inline 11-bug_tracking.Rmd:77]:
```suggestion
2. Prior to reporting the issue, please go through the $\textit{closed}$ issues to check if the bug is already reported and fixed. Otherwise, please check in the $\textit{open}$ issues to see if it is reported. If it is not found at either of these places, then please submit a new issue.

```

File: 11-bug_tracking.Rmd
Comment (willingc) [inline 11-bug_tracking.Rmd:83]:
```suggestion
2. The bug report for $\textsf{R}$ Core-supported packages and/or their documentations can be submitted either to $\textsf{R}$'s [Bugzilla](https://bugs.r-project.org/bugzilla/). In the future, we hope to have an option to report an issue to the [GitHub Mirror of R svn server](https://github.com/r-devel/r-svn/issues).

```

**Review by willingc — COMMENTED** [review]
Looking good @SaranjeetKaur. I've made a few suggestions along with those that @llrs mentioned.

File: 11-bug_tracking.Rmd
Comment (SaranjeetKaur) [reply 11-bug_tracking.Rmd:88]: This looks like a good way to mention the GitHub Mirror of R svn server. Let's see what more suggestions come for this point.

File: 11-bug_tracking.Rmd
Comment (hturner) [inline 11-bug_tracking.Rmd:101]: This is probably out of date now - it is less common to have to email the maintainer these days. The best practice is to provide a url where bug reports can be submitted (an issues tracker on GitHub/GitLab/R-Forge). So I would say to look at the package DESCRIPTION file first (e.g. using `packageDescription()`) to check if a url is provided in the `BugReports` field. If that is not available. Then contact the maintainer.

File: 11-bug_tracking.Rmd
Comment (hturner) [reply 11-bug_tracking.Rmd:103]: Interesting! Also I would not enumerate these paragraphs, the `bug.report()` function is a helper to follow the process I described above, so it is the same point, not a sequence of points.

File: 11-bug_tracking.Rmd
Comment (hturner) [inline 11-bug_tracking.Rmd:115]: Yes, this probably the main area where we will end up with too many versions of documentation. In time, this guide might replace the guidance on the Bug Reporting page of the R project site, but it is fine to cross-link for now while we are developing the guide.

File: 11-bug_tracking.Rmd
Comment (hturner) [reply 11-bug_tracking.Rmd:118]: Yes. This would be good to discuss e.g. at the RCWG and get input from R core members.

File: 11-bug_tracking.Rmd
Comment (hturner) [reply 11-bug_tracking.Rmd:125]: This should be Bugzilla vs GitHub mirror here.

**Review by hturner — CHANGES_REQUESTED** [review]
As Lluis mentioned, it is difficult to get a good overview of the documentation at this stage. So it is hard to get an idea if the logical flow is right or if some information is redundant. But I have made some comments on specific points for you to resolve and I think if address these along with the previous comments made you will have a good first version to merge.

File: 11-bug_tracking.Rmd
Comment (hturner) [reply 11-bug_tracking.Rmd:118]: Yes. This would be good to discuss e.g. at the RCWG and get input from R core members.

File: 11-bug_tracking.Rmd
Comment (hturner) [reply 11-bug_tracking.Rmd:112]: Linking to the SO question is a good idea - as it is a community wiki post, that is quite neutral and gives generally useful advice. We might want to emphasise specific points - e.g. use inbuilt data sets as far as possible. Again, something to get feedback from R Core on.

File: 11-bug_tracking.Rmd
Comment (gmbecker) [reply 11-bug_tracking.Rmd]: I agree with @llrs, I definitely do not feel that this guide should suggest reporting, tracking, or commenting on bugs in R via the github mirror. My understanding is the same as his.

File: 11-bug_tracking.Rmd
Comment (gmbecker) [reply 11-bug_tracking.Rmd:125]: I have not read the whole document yet so it may be mentioned elsewhere, but every comment generates an email to every R-core member (or at least ones that don't have notifications disabled), and so care should be taken when commenting on issues in a couple of different cases, which include a) issues that have already been considered and closed, and b) commenting in pure agreement without adding anything beyond that to a discussion (+1 type posts which are perfectly acceptable in other contexts).

**Comment by hturner** [comment]
The changes all look fine. Just one more thing before we merge: can you use markdown syntax to mark text as bold/italic, etc. I suggest using `**bold**` for bold text and `_italic_` for italic text so that they're clearly distinct.

**Review by hturner — APPROVED** [review]

**Pull request merged**
